jellyfin_kodi/objects/obj.py

```diff
@@ -125,11 +125,10 @@ def __recursiveloop__(self, obj, keys):
         obj = self.__recursive__(obj, first)
 
         if obj:
-            if rest:
-                for item in obj:
+            for item in obj:
+                if rest:
                     self.__recursiveloop__(item, rest)
-            else:
-                for item in obj:
+                else:
```

Comment:
This looks like a potential logic change, needs double check (we really need to get good test coverage going.)